### PR TITLE
Add com.w1hkj.fldigi

### PR DESCRIPTION
**Pull request by lu9dce**
Fldigi (short for Fast light digital) is a free and open-source program which allows an ordinary computer's sound card to be used as a simple two-way data modem. The software is mostly used by amateur radio operators who connect the microphone and headphone connections of an amateur radio SSB transceiver or an FM two way radio to the computer's headphone and microphone connections, respectively.

FLDigiWindowHtml

This interconnection creates a "sound card defined radio" whose available bandwidth is limited by the sound card's sample rate and the external radio's bandwidth.

Such communications are normally done on the shortwave amateur radio bands in modes such as PSK31, MFSK, RTTY, Olivia, and CW (morse code). Increasingly, the software is also being used for data on VHF and UHF frequencies using faster modes such as 8-PSK.

Using this software, it is possible for amateur radio operators to communicate worldwide while using only a few watts of RF power.

Fldigi software is also used for amateur radio emergency communications when other communication systems fail due to natural disaster or power outage. Transfer of files, emails, and FEMA ICS forms are possible using inexpensive radio hardware

### CODE REVIEW

**Comment by bilelmoussaoui** [comment]
bot, build org.freedesktop.fldigi

File: org.freedesktop.fldigi.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+app-id: org.freedesktop.fldigi
```
Comment (bilelmoussaoui) [inline org.freedesktop.fldigi.yml:1]: The same question as your old PR, I don't think you can use org.freedesktop as a prefix for your app-id because it's not part of freedesktop ?

**Comment by lu9dce** [comment]
fldigi

Perfect test in slackware 14.2

File: org.freedesktop.fldigi.yml (hunk 1 above)
Comment (lu9dce) [reply org.freedesktop.fldigi.yml:1]: Tell me how it should be. This is my first flatpak and I don't understand English. I try to translate all the manuals I can, but there are things that escape when translating it. Help me with this to correct it.
I have many programs that are used for hobby radio that I want
include
I do this because many are older people and are starting with linux, when using flatpak the programs run practically in any distro
I would appreciate your help, thank you

Comment (X-m7) [reply org.freedesktop.fldigi.yml:1]: It seems like the program's main website is at http://www.w1hkj.com/, so com.w1hkj.fldigi would be appropriate (not sure if digits would be valid in app ids though).

Comment (lu9dce) [reply org.freedesktop.fldigi.yml:1]: ok tks -- 
should I rename the files too?
Captura de pantalla_2020-11-29_18-44-14

Comment (X-m7) [reply org.freedesktop.fldigi.yml:1]: Yes, make sure to update the file names and build commands in the yml file too.

Comment (lu9dce) [reply org.freedesktop.fldigi.yml:1]: ok perfect tks

File: com.w1hkj.fldigi.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+sdk: org.freedesktop.Sdk
+command: fldigi
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=all
+  - --share=network
+  - --socket=pulseaudio
+
```
Comment (barthalion) [inline com.w1hkj.fldigi.yml:13]: You should add a `cleanup` list, with at least `*.a` and `/include`.

**Comment by barthalion** [comment]
Have you asked upstream developer if they would like to maintain their app on Flathub?

File: com.w1hkj.fldigi.desktop [hunk 3]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
```
Comment (AsciiWolf) [inline com.w1hkj.fldigi.desktop:1]: Feel free to also add a desktop file for Flarq. Flatpak supports multiple desktop files for one Flatpak. For example, see OpenMW.

Comment (AsciiWolf) [reply com.w1hkj.fldigi.desktop:1]: By the way, if desktop files are available in the tar.gz archive, you can use them for there. You don't need to include them in the Flatpak Git repo.

File: com.w1hkj.fldigi.yml (hunk 2 above)
Comment (lu9dce) [reply com.w1hkj.fldigi.yml]: ok

File: com.w1hkj.fldigi.desktop (hunk 3 above)
Comment (lu9dce) [reply com.w1hkj.fldigi.desktop:1]: this is not clear to me
if the program brings the .desktop
so I shouldn't recreate it
or you have to specify something in the manifest

**Comment by lu9dce** [comment]
good added the flarq
I didn't find another way to use the .desktop
check several pull of other programs and can not find anything .. keep looking
There are also other programs made by the same developer I will see to go up as I can make the manifests better
With other kinds of programs for amateur radio mainly made in qt I find it difficult to search for libraries in / usr / lib and / usr / local / lib
for example the wsjtx
but I'll leave it later
unfortunately my lack of knowledge of the English language hurts me .. but with 50 years it does not give me the head to learn another language

**Comment by AsciiWolf** [comment]
> unfortunately my lack of knowledge of the English language hurts me .. but with 50 years it does not give me the head to learn another language

No problem, we are here to help you. I will try to do my best when I have some free time...

File: com.w1hkj.fldigi.desktop (hunk 3 above)
Comment (lu9dce) [reply com.w1hkj.fldigi.desktop:1]: fldigi has no icon in png usa xpm
I tried several ways to do it but it didn't work
installing on a local pc
fldigi andava but did not show the icon
so how are you at the moment it's fine

**Comment by lu9dce** [comment]
@bilelmoussaoui all running and testing ... send the bot tks

**Comment by bilelmoussaoui** [comment]
You can trigger the bot as well

**Comment by lu9dce** [comment]
how is it done?

**Comment by lu9dce** [comment]
bot, build org.freedesktop.fldigi

**Comment by lu9dce** [comment]
bot, build com.w1hkj.fldigi

**Comment by lu9dce** [comment]
bot, build com.w1hkj.fldigi

File: com.w1hkj.fldigi.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  do frequency measurement tests.
+  </p>
+ </description>
+ <screenshots>
+  <screenshot type="default">
+   <image>http://www.w1hkj.com/FldigiHelp/FLDigiWindowHtml.png</image>
+  </screenshot>
+ </screenshots>
+ <releases>
+   <release version="4.1.16" date="2020-11-1"/>
```
Comment (barthalion) [inline com.w1hkj.fldigi.metainfo.xml:26]:
```suggestion
   <release version="4.1.16" date="2020-11-13"/>
```

File: com.w1hkj.fldigi.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
+app-id: com.w1hkj.fldigi
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+sdk: org.freedesktop.Sdk
+command: fldigi
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
```
Comment (barthalion) [inline com.w1hkj.fldigi.yml:9]:
```suggestion
  - --socket=x11
```

**Comment by lu9dce** [comment]
bot, build com.w1hkj.fldigi

File: com.w1hkj.fldigi.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+sdk: org.freedesktop.Sdk
+command: fldigi
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --device=all
+  - --share=network
+  - --socket=pulseaudio
```
Comment (AsciiWolf) [inline com.w1hkj.fldigi.yml:11]: No settings seem to be saved. You should add
```
   - --persist=.fldigi
   - --persist=.fltk
   - --persist=.nbems
```
to the finish-args to make these directories persistent.

**Review by AsciiWolf — CHANGES_REQUESTED** [review]

**Comment by lu9dce** [comment]
**well I found out that the app doesn't save user settings
surely for lack of any permission
I'll see what permission you need to modify**

File: com.w1hkj.fldigi.yml (hunk 6 above)
Comment (lu9dce) [reply com.w1hkj.fldigi.yml:11]: ok

**Comment by AsciiWolf** [comment]
The three persistent directories mentioned in my previous comment should be sufficient. :)

File: com.w1hkj.fldigi.yml (hunk 6 above)
Comment (lu9dce) [reply com.w1hkj.fldigi.yml:11]: ahhh I just realized when I provided it on my machine .. thanks .. correcting

Comment (lu9dce) [reply com.w1hkj.fldigi.yml:11]: the same for flcluster .. because it doesn't save

**Comment by AsciiWolf** [comment]
By the way, consider re-adding the Flarq desktop/icon files. They should work fine, you just have to give them a proper name, see for example this: https://github.com/flathub/org.openmw.OpenMW/blob/master/org.openmw.OpenMW.yaml#L150

So, `com.w1hkj.fldigi.flarq.desktop` should work fine. :)

**Comment by lu9dce** [comment]
> By the way, consider re-adding the Flarq desktop/icon files. They should work fine, you just have to give them a proper name, see for example this: https://github.com/flathub/org.openmw.OpenMW/blob/master/org.openmw.OpenMW.yaml#L150
> 
> So, `com.w1hkj.fldigi.flarq.desktop` should work fine. :)

add the icon but it won't run
.desktop with everything
flatpak run .... just run the fldigi
I don't know how to add it

**Comment by lu9dce** [comment]
checking flarq

**Comment by AsciiWolf** [comment]
> add the icon but it won't run
> .desktop with everything
> flatpak run .... just run the fldigi
> I don't know how to add it

No, there don't have to be "flatpak run" in the desktop file, just `Exec=flarq`. If you want to run it from a terminal, you can use: `flatpak run --command=flarq com.w1hkj.fldigi`

**Comment by AsciiWolf** [comment]
bot, build com.w1hkj.fldigi

**Comment by lu9dce** [comment]
> No, there don't have to be "flatpak run" in the desktop file, just `Exec=flarq`. If you want to run it from a terminal, you can use: `flatpak run --command=flarq com.w1hkj.fldigi`

ahh ok

**Comment by lu9dce** [comment]
/usr/include/c++/10.2.0/complex:1227:12: note: ‘cmplx’ {aka ‘struct std::complex<double>’} declared here

apparently it is an incompatibility with the c ++ version

**Comment by AsciiWolf** [comment]
I have tried basic functionality of Fldigi from this Flatpak and everything seems to work fine now.

So, LGTM, except the missing Flarq desktop file.

**Comment by lu9dce** [comment]
@AsciiWolf ok
test flarq

**Comment by lu9dce** [comment]
bot, build com.w1hkj.fldigi

**Comment by lu9dce** [comment]
what I most wanted to put in flatpak was the wsjtx
that the number one program today for radio amateurs
it's done in qt this is easy
the problem is that it will search all the libraries
to / usr / bin, and from what I saw it doesn't work in flatpak
I'm still going to do as far as I go
and I will upload it to see if others can help resolve the issue

File: com.w1hkj.fldigi.yml [hunk 7]
```diff
@@ -11,7 +11,6 @@ finish-args:
   - --socket=pulseaudio
   - --persist=.fldigi
   - --persist=.fltk
-  - --persist=.nbems
```
Comment (AsciiWolf) [inline com.w1hkj.fldigi.yml]: Why did you delete the `.nbems` persistent directory? It was not used by flcluster, however it is used by fldigi.

File: com.w1hkj.fldigi.yml [hunk 8]
```diff
@@ -44,13 +43,19 @@ modules:
   - name: metainfo
     buildsystem: simple
     build-commands:
-        - install -D com.w1hkj.fldigi.metainfo.xml  /app/share/metainfo/com.w1hkj.fldigi.metainfo.xml
+        - install -D com.w1hkj.fldigi.metainfo.xml /app/share/metainfo/com.w1hkj.fldigi.metainfo.xml
         - install -D com.w1hkj.fldigi.png /app/share/icons/hicolor/128x128/apps/com.w1hkj.fldigi.png
+        - install -D com.w1hkj.flarq.png  /app/share/icons/hicolor/128x128/apps/com.w1hkj.flarq.png
         - install -D com.w1hkj.fldigi.desktop /app/share/applications/com.w1hkj.fldigi.desktop
+        - install -D com.w1hkj.flarq.desktop  /app/share/applications/com.w1hkj.flarq.desktop
```
Comment (AsciiWolf) [inline com.w1hkj.fldigi.yml:50]: This is not a correct name. The correct name should be `com.w1hkj.fldigi.flarq.desktop` (and same thing, just with the .png suffix for the icon name), as I mentioned before.

**Review by AsciiWolf — CHANGES_REQUESTED** [review]

File: com.w1hkj.flarq.desktop [hunk 9]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=Flarq
+GenericName=Fast Light Automatic Repeat reQuest
+Comment=Fast Light Automatic Repeat reQuest
+Exec=flatpak run --command=flarq com.w1hkj.fldigi
+Icon=com.w1hkj.flarq.png
```
Comment (AsciiWolf) [inline com.w1hkj.flarq.desktop:6]: Remove the `.png` suffix.

File: com.w1hkj.flarq.desktop [hunk 10]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=Flarq
+GenericName=Fast Light Automatic Repeat reQuest
+Comment=Fast Light Automatic Repeat reQuest
+Exec=flatpak run --command=flarq com.w1hkj.fldigi
```
Comment (AsciiWolf) [inline com.w1hkj.flarq.desktop:5]: Change this to `Exec=flarq`.

**Review by AsciiWolf — CHANGES_REQUESTED** [review]

File: com.w1hkj.fldigi.yml (hunk 8 above)
Comment (lu9dce) [reply com.w1hkj.fldigi.yml:50]: ok

File: com.w1hkj.flarq.desktop (hunk 9 above)
Comment (AsciiWolf) [reply com.w1hkj.flarq.desktop:6]: Also, don't forget to change it to `com.w1hkj.fldigi.flarq` after you rename the desktop/icon files.

**Comment by AsciiWolf** [comment]
https://github.com/flathub/flathub/pull/1959#discussion_r533576746
https://github.com/flathub/flathub/pull/1959#discussion_r533578916
https://github.com/flathub/flathub/pull/1959#discussion_r533579788

Still waiting to be fixed. Otherwise LGTM.

File: com.w1hkj.fldigi.yml (hunk 7 above)
Comment (AsciiWolf) [reply com.w1hkj.fldigi.yml]: Consider reverting this change.

**Comment by AsciiWolf** [comment]
Two more fixes remaining. :-)

**Comment by lu9dce** [comment]
bot, build com.w1hkj.fldigi

**Comment by lu9dce** [comment]
bot, build com.w1hkj.fldigi

File: com.w1hkj.fldigi.yml (hunk 7 above)
Comment (AsciiWolf) [reply com.w1hkj.fldigi.yml]: @lu9dce **^** One more thing to fix. ;-)

Comment (lu9dce) [reply com.w1hkj.fldigi.yml]: - --persist = .nbems
I already took it out
add it again?

Comment (AsciiWolf) [reply com.w1hkj.fldigi.yml]: Yes.

**Comment by lu9dce** [comment]
bot, build com.w1hkj.fldigi

**Comment by AsciiWolf** [comment]
Everything seems to be fine now, thanks!

**Comment by lu9dce** [comment]
very good .. thank you all for helping 
----------------
                               **sharing knowledge is a heritage of humanity**

**Comment by AsciiWolf** [comment]
By the way, if you would like to make Flatpaks for other amateur radio apps, one thing that is missing on Flathub are SDR apps. For example, SDR Angel [already has](https://github.com/f4exb/sdrangel/tree/master/flatpak) Flatpak, it's just not on Flathub. However, it needs to be ported to a latest KDE Sdk/Runtime. Last time I tired it, there were some issues and I did not have enough time to try fixing them. Maybe, you will have more luck with it. :-) Also, there is another issue with SDR software in Flatpak: Many SDRs require custom udev rules, so for example for RTL-SDR, users still have to manually install rtl-sdr package or copy the udev rules from this package to their _/etc/udev/rules.d/_ directory to be able to use Flatpak SDR Angel with RTL-SDR.

[WSJT-X](https://www.physics.princeton.edu/pulsar/k1jt/wsjtx.html) Flatpak would also be great.

Or another interesting ham radio apps worth having a Flatpak version:
[Flrig](http://www.w1hkj.com)
[QSSTV](http://users.telenet.be/on4qz/)
[Xastir](https://xastir.org/index.php/Main_Page)
[Qtel](https://www.svxlink.org)
[CHIRP](https://chirp.danplanet.com/projects/chirp/wiki/Home)

**Comment by lu9dce** [comment]
ok understood
yes now I'm about to upload tqsl almost ready
I am improving the manifest to try to obtain and avoid uploading the icon and .desktop
and I am seeing xastir (it is giving me problems) when I have it more or less walking I will raise it so give me a hand
Qtel would be good .. and the one I want to upload the most is wsjtx
which gives me problems with the / usr folder

**Comment by AsciiWolf** [comment]
> and the one I want to upload the most is wsjtx
> which gives me problems with the / usr folder

I am not sure if this could help fix your issue, but I have found that WSJT-X supports these cmake options:
```
-D CMAKE_PREFIX_PATH:PATH=
-D CMAKE_INSTALL_PREFIX=
```

However, CMAKE_INSTALL_PREFIX should be defined by default when using cmake/cmake-ninja buildsystem and CMAKE_PREFIX_PATH should not be needed.

What exact issue do you have when trying to compile the WSJT-X Flatpak?

**Comment by barthalion** [comment]
Unsure if it failed randomly or for a good reason, let's try again.

/merge @AsciiWolf

**Comment by barthalion** [comment]
Merged manually.
